### PR TITLE
fix: add bounds validation to BattleEngine.deserialize()

### PR DESCRIPTION
**Pull request by uehlbran**
## Summary

- Add `validateRestoredState()` check in `deserialize()` that rejects negative `turnNumber` and negative `currentHp` values
- Add 2 new tests for these validation edge cases
- Lightweight hardening — rejects clearly invalid state without being overly strict

## Test plan

- [x] All 26 deserialize tests pass (24 existing + 2 new)
- [x] `npm run typecheck` passes
- [x] Biome clean

Closes: N/A

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Bug Fixes**
  * Added validation when restoring saved battles to prevent loading corrupted or invalid data. The system now rejects invalid turn numbers and out-of-range health values, protecting against compromised save files affecting gameplay.

### CODE REVIEW

File: packages/battle/src/engine/BattleEngine.ts [hunk 1]
```diff
@@ -1540,6 +1541,24 @@ export class BattleEngine implements BattleEventEmitter {
...
+    for (const side of state.sides) {
+      for (const pokemon of side.team) {
+        if (typeof pokemon.currentHp !== "number" || pokemon.currentHp < 0) {
+          throw new Error(
+            `BattleEngine.deserialize: Pokemon ${pokemon.uid} has invalid currentHp ${pokemon.currentHp}`,
+          );
+        }
+      }
+    }
+  }
```
[the remark this answers is not in the document]
Comment (uehlbran) [reply packages/battle/src/engine/BattleEngine.ts]: NaN cannot appear in the deserialized state because `JSON.parse` converts NaN to null. The typeof check catches null. Infinity is also not valid JSON. The validation is correct for the JSON deserialization path.
